### PR TITLE
Mem 49 make very simple login screen

### PR DESCRIPTION
**Pull request by neiljaywarner**
# Authentication and Authorization Features

## Summary
This PR adds full authentication and authorization capabilities to the Memverse app, including secure token storage and management for API requests. The implementation includes a complete login flow and integration with the Memverse API.

## Key Features
- OAuth 2.0 authentication implementation with Memverse API
- Secure token storage using Flutter Secure Storage
- Token management for API requests with proper Authorization headers
- Login UI with form validation and error handling
- Auth state management using Riverpod
- Integration tests verifying end-to-end login flow

## Technical Details
- Added `AuthToken` domain model to represent and manage OAuth tokens
- Created `AuthService` for handling login, token storage, and session management
- Added comprehensive Riverpod providers for auth state management:
  - `authStateProvider` to maintain authentication state
  - `accessTokenProvider` and `bearerTokenProvider` for API authorization
- Implemented secure storage of tokens with `FlutterSecureStorage`
- Added `LoginPage` UI for user credentials entry
- Added `AuthWrapper` component to handle authenticated/unauthenticated states

## Testing
- Added integration test verifying full login flow with real API credentials
- Included test setup documentation for environment variable configuration
- Tested token persistence and retrieval on app restart
- Verified token format for API authorization headers

## Dependencies Added
- flutter_secure_storage: For securely storing authentication tokens
- dio: For improved API request handling with good error management

## Next Steps
- Add automatic token refresh capabilities
- Implement logout functionality in the UI
- ?? If this is worth itAdd remember me option for persistent logins
- Autofill and show password "eye"

The authentication system now successfully saves tokens securely and allows the app to make authorized API requests to the Memverse backend.

//Update also got this ai generated markdown below for revisions
# Configuration And Testing Improvements

## Summary
This PR introduces a robust configuration management system and enhances the testing infrastructure, making the codebase more reliable and easier to maintain. The implementation ensures that required environment variables are properly validated and provides clear error messages when configuration is missing.

### 1. Centralized Configuration Management
- Created a shared configuration system with a single source of truth
- Added `project_config.sh` to store shared configuration values
- Implemented a system for extracting configuration values in GitHub Actions workflows
- Eliminated duplicate configuration values across build scripts and CI/CD pipelines

### 2. Enhanced Error Handling
- Added proper validation for required environment variables
- Created user-friendly error screens when critical configuration is missing
- Provided clear instructions on how to run the app with proper configuration
- Improved logging for configuration-related issues

### 3. Code Quality Improvements
- Fixed analyzer issues related to const usage and redundant arguments
- Improved type safety throughout the codebase
- Enhanced coverage thresholds and excluded generated files
- Made all tests pass with 99.3% coverage

### 4. Build Process Enhancements
- Improved `check_before_commit.sh` script with better coverage validation
- Enhanced GitHub Actions workflow to extract configuration from source code
- Added consistent code formatting rules across environments
- Fixed issues with test coverage calculation

## Technical Details
- Created ConfigurationErrorWidget to provide helpful error messages
- Modified bootstrap.dart to validate CLIENT_ID early in the startup process
- Improved error handling and error UI for missing configuration
- Fixed issues with argument values in auth providers
- Implemented a more robust method to check coverage thresholds

## Testing
- All tests now pass with 99.3% coverage
- Added better error handling and validation in tests
- Fixed issues with test setup documentation

## Next Steps
- Consider adding more configuration values to the centralized system
- Improve test coverage for edge cases in configuration handling
- Enhance error messages with more context-sensitive information
- Consider adding automated validation for the config file format

This PR ensures better consistency between local development and CI/CD environments by having a single source of truth for configuration values like coverage thresholds.

### CODE REVIEW

File: lib/src/features/auth/presentation/providers/auth_providers.dart [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  Future<void> logout() async {
+    try {
+      state = state.copyWith(isLoading: true);
+      await _authService.logout();
+      state = const AuthState();
+    } catch (e) {
+      state = state.copyWith(isLoading: false, error: e.toString());
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (neiljaywarner) [reply lib/src/features/auth/presentation/providers/auth_providers.dart:118]: i guess let's skip this for now

**Review by njwandroid — APPROVED** [review]

File: lib/src/features/verse/data/verse_repository.dart [hunk 2]
```diff
@@ -102,30 +104,34 @@ class LiveVerseRepository implements VerseRepository {
...
       // During tests, Dio is usually mocked so we don't need a real token
-      if (_privateToken.isEmpty && !isInTestMode && !_dio.isTestMockDio) {
-        throw Exception(
-          'No API token provided. Run with --dart-define=MEMVERSE_API_TOKEN=your_token',
-        );
+      final token = _ref.read(accessTokenProvider);
+      final bearerToken = _ref.read(bearerTokenProvider);
+      final isTestMockDio = _dio.isTestMockDio;
+      if (token.isEmpty && !isInTestMode && !isTestMockDio) {
+        throw Exception('No API token provided. Please login to get a valid token');
```
[the remark this answers is not in the document]
Comment (njwandroid) [reply lib/src/features/verse/data/verse_repository.dart:117]: don't worry about this for now

**Review by njwandroid — COMMENTED** [review]
approved for now, will give more review later

**Review by njwandroid — APPROVED** [review]

**Review by ericlsimplifi — COMMENTED** [review]
A bit too long for my limited time sorry. :rubber_stamp:

**Comment by neiljaywarner** [comment]
Totally!

I'm more meant it as an FYI Or a someday maybe.

With where I'm at with my flutter expertise and where people are with their
time, I'm not really expecting to slow down my work for approvals to be
honest, but only vice versa at least for now

. Sorry about that

....sent from my phone
